runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1455,14 +1455,310 @@ TR::Register *J9::X86::TreeEvaluator::newEvaluator(TR::Node *node, TR::CodeGener
+    *    | m reference array slots |  | spine padding bytes + header bytes + m * reference bytes
+    *    |-------------------------|  |
+    *    | spine array header      | /
+    *    |-------------------------|
+    *    | padding for alignment   | vmThread->heapAlloc padding bytes
+    *    |-------------------------|
+    *
+    * If the whole allocation won't fit between vmThread->heapAlloc and vmThread->heapTop,
+    * call the snippet instead which will handle the heap overflow.
+    *
+    * After allocating, initialise
+    *   - Spine array header with class and size
+    *   - Spine array slots with references to leaf arrays
+    *   - Leaf array headers
+    */
+
+   // alignment requirement
+   int32_t alignmentInBytes = TR::Compiler->om.getObjectAlignmentInBytes();
+   // a length>0 array object would *not* require alignment if both a single element
+   // and the header are already the exact multiple of alignment; otherwise alignment is needed
+   uintptr_t contiguousArrayHeaderSize = TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
+   bool needsAlignLeaf = (OMR::align(leafArrayElementSize, alignmentInBytes) != leafArrayElementSize);
+   bool needsAlignHeader = (contiguousArrayHeaderSize
+      != OMR::align(contiguousArrayHeaderSize, alignmentInBytes));
+
+   bool use64BitClasses = !TR::Compiler->om.generateCompressedObjectHeaders();
+
+   bool skipZeroInit = fej9->tlhHasBeenCleared() || node->canSkipZeroInitialization();
+   TR::Register *zeroReg = skipZeroInit ? NULL : cg->allocateRegister();
+   TR::Register *tmpReg = skipZeroInit ? NULL : cg->allocateRegister();
+
+   uintptr_t classOffset = TR::Compiler->om.offsetOfObjectVftField();
+   uintptr_t sizeOffset = fej9->getOffsetOfContiguousArraySizeField();
+
+   // load dimensions and class
+   TR::Register *dimsPtrReg = cg->evaluate(sizeArrNode);
+   TR::Register *secondDimLenReg = cg->allocateRegister();
+   TR::Register *classReg = cg->longClobberEvaluate(classNode);
+   TR::Register *firstDimLenReg = cg->allocateRegister();
+
+   static const bool breakOnInlineGenerate2DArray = feGetEnv("TR_BreakOnInlineGenerate2DArray") != NULL;
+   if (breakOnInlineGenerate2DArray)
+      generateInstruction(TR::InstOpCode::INT3, node, cg);
+
+   generateRegMemInstruction(TR::InstOpCode::MOVSXReg8Mem4, node, secondDimLenReg, generateX86MemoryReference(dimsPtrReg, 0, cg), cg);
+   generateRegMemInstruction(TR::InstOpCode::MOVSXReg8Mem4, node, firstDimLenReg, generateX86MemoryReference(dimsPtrReg, 4, cg), cg);
+
+   // calculate spine array size = header bytes + first dim * reference bytes
+   TR::Register *spineSizeReg = cg->allocateRegister();
+   int32_t spineArraySlotShift = trailingZeroes(TR::Compiler->om.sizeofReferenceField();
+   generateRegMemInstruction(TR::InstOpCode::LEARegMem(), node, spineSizeReg,
+                             generateX86MemoryReference(NULL, firstDimLenReg, spineArraySlotShift), contiguousArrayHeaderSize, cg), cg);
```

Comment:
Trying to cherry-pick this change, I’m seeing this error building with it:

```
no matching function for call to 'generateX86MemoryReference(NULL, TR::Register*&, int32_t&)'
```

I’m not sure whether this is caused by some API change that has happened since this pull request was opened.

Reply:
I must have moved some parentheses while dealing with long lines, fixed now